filter/src/lib.rs

```diff
@@ -49,6 +51,67 @@ enum Do {
     Httpbin(String),
 }
 
+impl fmt::Debug for Do {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        return write!(
+            f,
+            "{}",
+            match self {
+                Do::Fail => "Do.Fail",
+                Do::Redirect(_) => "Do.Redirect",
+                Do::Body(_) => "Do.Body",
+                Do::Httpbin(_) => "Do.Httpbin",
+            }
+        );
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+struct RequestCount {
+    request_count: u64,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+struct RequestEvent {
+    request_key: String,
+}
+
+impl Filter {
+    fn bump_request_ct(&self, action: &Option<Do>) -> u64 {
+        let evt_name = format!("envoy.playground.request_ct.{:?}", action);
+        let queue_id = match self.resolve_shared_queue("my_vm_id", &self.config.channel_name) {
```

Comment:
Generally, the pattern:

```
let foo = match bar {
  Some(thing) => thing,
  None => panic!("boom"),
};
```

or

```
let foo = match bar {
  Ok(thing) => thing,
  Err(_) => panic!("boom"),
};
```

is spelled `foo.expect("boom")`

Reply:
ohhhh, that's handy. I'll give it a go!